### DISCUSSION TITLE
클로저와 관련된 메모리 관리는 어떻게 이루어지나요?

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#92] [클로저]
- 클로저와 관련된 메모리 관리는 어떻게 이루어지나요?

**Comment by pipisebastian** [comment]
> * 클로저와 관련된 메모리 관리는 어떻게 이루어지나요?

- 내부함수가 외부함수의 렉시컬 환경을 참조하고 있기 때문에, 가비지 컬렉션의 대상이 되지 않고, 외부함수가 종료되더라도, 외부함수의 렉시컬 환경 - enviroment record 가 살아있게 됩니다.
- 클로저의 메모리 정보는, 렉시컬 환경을 다 기억하는 것이 아닌, 필요한 식별자, 즉 참조중인 식별자만 기억합니다.

**Comment by publdaze** [comment]
> - 클로저와 관련된 메모리 관리는 어떻게 이루어지나요?

- 중첩함수가 외부함수의 식별자를 참조하고 있을 경우 해당 렉시컬 환경을 기억해야 하기 때문에 가비지 컬렉션의 대상이 되지 않고 메모리에 남아있음
- 만약 참조하고 있는 중첩함수의 실행이 끝나고 더이상 해당 렉시컬 환경을 참조하고 있는 대상이 없다면 그제서야 가비지 컬렉션에 의해 메모리 해제될 듯
